includes/Core/Site_Health/Tags_Placement.php

```diff
@@ -0,0 +1,257 @@
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Site_Health;
+
+use Google\Site_Kit\Core\Modules\Modules;
+use Google\Site_Kit\Core\Modules\Module_With_Tag;
+use Google\Site_Kit\Core\Modules\Tags\Module_Tag_Matchers;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\REST_API\REST_Route;
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+use WP_REST_Server;
+
+/**
+ * Class for integrating status tab information with Site Health.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Tags_Placement {
+
+	use Method_Proxy_Trait;
+
+	/**
+	 * Modules instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Modules
+	 */
+	private $modules;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Modules $modules Modules instance.
+	 */
+	public function __construct( Modules $modules ) {
+		$this->modules = $modules;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_filter(
+			'googlesitekit_rest_routes',
+			function ( $rest_routes ) {
+				$health_check_routes = $this->get_rest_routes();
+
+				return array_merge( $rest_routes, $health_check_routes );
+			}
+		);
+
+		add_filter(
+			'site_status_tests',
+			function ( $tests ) {
+				global $wp_version;
+
+				if ( version_compare( $wp_version, '5.6', '<' ) ) {
+					$tests['direct']['tag_placement'] = array(
+						'label' => __( 'Tags Placement', 'google-site-kit' ),
+						'test'  => $this->get_method_proxy( 'tags_placement_test' ),
+					);
+
+					return $tests;
+				}
+
+				$tests['async']['tag_placement'] = array(
+					'label'             => __( 'Tags Placement', 'google-site-kit' ),
+					'test'              => rest_url( 'google-site-kit/v1/core/site/data/tags-placement-test' ),
+					'has_rest'          => true,
+					'async_direct_test' => $this->get_method_proxy( 'tags_placement_test' ),
+				);
+
+				return $tests;
+			}
+		);
+	}
+
+	/**
+	 * Gets all REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return REST_Route[] List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		return array(
+			new REST_Route(
+				'core/site/data/tags-placement-test',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => $this->get_method_proxy( 'tags_placement_test' ),
+						'permission_callback' => function () {
+							return current_user_can( Permissions::SETUP );
+						},
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Checks if the modules tags are placed on the website.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Site health status results.
+	 */
+	protected function tags_placement_test() {
+		global $wp_version;
+
+		$result = array(
+			'label'   => __( 'Tags Placement', 'google-site-kit' ),
+			'status'  => 'recommended',
```

Comment:
This status shouldn't be recommended by default. See https://github.com/google/site-kit-wp/issues/5695#issuecomment-1896322675